src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -334,6 +335,16 @@ ResetHideShardsDecision(void)
 static bool
 ShouldHideShardsInternal(void)
 {
+	if (MyBackendType != B_BACKEND)
+	{
```

Comment:
one question here is: does this cause commands executed via pg_cron to behave differently from commands executed by the user?

we do not run pg_cron in background worker mode, but we're likely to forget about this if we change it in the future

Reply:
> one question here is: does this cause commands executed via pg_cron to behave differently from commands executed by the user?

Ah, currently yes. 

Not sure you like, but we can add exception for `pg_cron` and others:
```C
	if (MyBackendType == B_BG_WORKER &&
		/* pg_cron, maintenance daemon, citus_lock_acquire */
		!IsCitusBgWorker())
	{
		/*
		 * Citus bg workers should be indistinguishable from
		 * client backends. We let those bg workers
		 */
		return false;
	}
	else if (MyBackendType != B_BACKEND)
	{
		/*
		 * We are aiming only to hide shards from client
		 * backends, not backends like logical replication
		 * or walsender.
		 */
		return false;
	}
```

Where we can either hard code bgw prefixes for `IsCitusBgWorker` or add a GUC?

Reply:
Or, maybe we do not hide shards from any pg bg workers?:
```C

	if (MyBackendType == B_BG_WORKER &&
		/* pg_cron, maintenance daemon, citus_lock_acquire */
		IsPgBgWorker())
	{
		/*
		 * Citus bg workers should be indistinguishable from
		 * client backends. We let those bg workers
		 */
		return false;
	}
	else if (MyBackendType != B_BACKEND)
	{
		/*
		 * We are aiming only to hide shards from client
		 * backends, not backends like logical replication
		 * or walsender.
		 */
		return false;
	}
```

Reply:
With this, we still hide shards from extensions like pg_prewarm (if queries pg_class). But, as soon such extensions use `pgstat_report_appname ` such as `pg_cron`, they obey `citus.hide_shards_from_app_name_prefixes`

Seems like an expected behaviour from users perspective

Reply:
Is there a way to show shards in case of pg_prewarm by changing a config?

Reply:
In general, citus.override_table_visibility still controls all the behavior. When set to false, nothing is hidden. So, we can set it globally if really needed. Btw, I don't think `pg_prewarm` or any other contrib bg-worker queries `pg_class` currently. Mine was a more generic answer. 

``citus.hide_shards_from_app_name_prefixes`` or any pg_prewarm setting (or other such extensions) doesn't help.

 I think one problem is that many bg workers do not set the application name.  They mostly set `backend_type` (via `bgw_type`). That's why we cannot use `application_name` approach we have.

```C
                                        if (MyProcPid == PostmasterPid)
                                                backend_type_str = "postmaster";
                                        else if (MyBackendType == B_BG_WORKER)
                                                backend_type_str = MyBgworkerEntry->bgw_type;
                                        else
                                                backend_type_str = GetBackendTypeDesc(MyBackendType);
```

I considered:
1) rename `citus.hide_shards_from_app_name_prefixes` -> `citus.hide_shards_from_client_app_name_prefixes`
2) introduce `citus.hide_shards_from_bg_workers_type_prefixes`

That seemed too complex for a non-existing problem for now (and also with override_table_visibility hack if it really becomes an issue)